File: Scala/day02/main.scala
import scala.io.Source

val totalWrapping = Source.fromFile("input.txt").getLines().map { line =>
    val lwh = line.split("x").map(_.toInt)
    val Array(l, w, h) = lwh
    val minSide = lwh.sorted.take(2).product
    2 * (l * w + w * h + l * h) + minSide
}.sum
println(totalWrapping)

val totalLength = Source.fromFile("input.txt").getLines().map { line =>
    val lwh = line.split("x").map(_.toInt)
    val minPerimeter = 2 * lwh.sorted.take(2).sum
    lwh.product + minPerimeter
}.sum
println(totalLength)

File: Java/day02/Main.java
import java.io.File;
import java.nio.file.Files;
import java.util.stream.IntStream;
import java.util.stream.Stream;

public class Main {

    public static void main(String[] args) throws Exception {
        int totalWrapping = Files.lines(new File("input.txt").toPath())
                .map(line -> {
                    IntStream intStream = Stream.of(line.split("x")).mapToInt(Integer::valueOf);
                    int[] lwh = intStream.sorted().toArray();
                    int minSide = lwh[0] * lwh[1];
                    return 2 * (lwh[0] * lwh[1] + lwh[1] * lwh[2] + lwh[0] * lwh[2]) + minSide;
                }).reduce((integer, integer2) -> integer + integer2).get();
        System.out.println(totalWrapping);

        int totalLength = Files.lines(new File("input.txt").toPath())
                .map(line -> {
                    IntStream intStream = Stream.of(line.split("x")).mapToInt(Integer::valueOf);
                    int[] lwh = intStream.sorted().toArray();
                    int minPerimeter = 2 * (lwh[0] + lwh[1]);
                    return lwh[0] * lwh[1] * lwh[2] + minPerimeter;
                }).reduce((integer, integer2) -> integer + integer2).get();
        System.out.println(totalLength);
    }
}
